### DISCUSSION TITLE
[FEATURE REQUEST] Turn off comments

### DISCUSSION THREAD

**Discussion by atomicpoet** [mastodon/mastodon#21444] [General]
When writing a post, have the ability to turn off comments from others. 

This is useful for when you want to broadcast a message, but do not want to engage in a conversation or have that conversation hi-jacked by bad actors.

**Comment by hutchisr** [comment]
I don't think this is a good idea. The Fediverse is for discussions, not "influencer" posting which is the only use case I see for this.

**Reply by amdavidson** [reply]
Influencer posting tends to be entirely about the engagement from user: comments, likes, etc. 

This would be excellent for posting a notice or some other message that doesn’t warrant engagement or would beg the wrong kind of attention.

**Reply by JoshuaACasey** [reply]
I think it's a totally great idea. The "supposed" reasoning for Mastodon not having QTs is "to prevent harassment". The problem is that harassment is not limited to QTs. It can happen (and mostly does) in replies, which is why being able to limit replies (either completing turning them off or limiting them only to people you follow) is such a great feature

**Reply by Joeviocoe** [reply]
> either completing turning them off or limiting them only to people you follow

For this to be fair (prevent the flood of broadcast only toots), users should have the option to filter such posts from their feeds.  Similar to how many people opt to not show bot accounts at all, they should be able to not see broadcasters.

**Comment by Freeplayg** [comment]
This would have to be implemented in ActivityPub first, wouldn't it?

**Reply by stefan-scholl** [reply]
If it is even possible. It's a feature only possible with a centralized instance.
Usenet showed that people can add anything they want to the meta data.

It would be possible as a suggestion that can get ignored. But you would need the support of all servers and clients for the Fediverse.

**Reply by unusualevent** [reply]
"if it's even possible" - that's like saying favoriting isn't possible.

**Reply by stefan-scholl** [reply]
https://github.com/mastodon/mastodon/blob/75bbe3b1b80207d9cab1a73a9fc38324ee6340d5/app/lib/activitypub/activity/like.rb#L9 is possible because of https://www.w3.org/TR/activitystreams-vocabulary/#dfn-like

How do you prevent comments on the protocol level?

**Reply by rbairwell** [reply]
How to prevent comments on the protocol level - well, if I understand ActivityPub/Mastodon enough (which might be not the case), comments (or at least a link to them) is sent back to the original server for threading purposes (so when someone views the thread, it knows where all the relevant posts are).
"All" the OP's server needs to do is ignore/disregard those comments. There is a possibility the comments will show up on the commenters server, but that should be short lived.
Ideally the protocol should say something like "comments_allowed:false" or have an empty comment url to inform UX that comments will be discarded and to save on network traffic.

**Comment by unusualevent** [comment]
two thoughts:

this could be respected by nice clients at the view toot and compose level, by hiding the reply icon and disabling the post icon if it's to a conversation with replies off

and

this could be respected by nice clients by hiding any toots that disobey the comment restriction - and not notifying anyone in the thread about them.

side note: this also could work for "only people I mention" or "only mutuals"

mostly for sealioning. http://wondermark.com/1k62/

**Comment by Orangestar12** [comment]
I don't know about having replies as a whole disabled (and I don't know about referring to them as "comments", it makes me feel like maybe the intention of the system is being misunderstood)

If someone's brigading a post, wouldn't it be easier to just block the bad actor? Or, if they're from a known bad instance, instance block them?

**Reply by unusualevent** [reply]
one... person? try... several. one person - that's reactive defense. proactive defense against abusers and malicious use would be a locked comment thread. Asking the victim to do the work of blocking /every single person/ that may brigade them (or their public health anouncement) is placing undo burden on someone when a simple (to the user) "lock comments on this post" (E.g., treat comments the same way you'd treat them if the person was blocked" button.

**Reply by neugens** [reply]
It surely seems to me a good feature to have.

**Reply by unusualevent** [reply]
I'm just saying, the task of allowlisting IPs is smaller than the task of "find out the IPs that are malicious"... and so-too with allowlisting actions for specific pre-vetted users.

**Comment by jonpincus** [comment]
Totally agree.  This is basic anti-harassment technology and the Twitter experience shows that it's helpful.  As @unusualevent describes, it's an important complement to blocking.

**Comment by narFnarF** [comment]
I'm not an expert, but based on my understanding of ActivityPub and Mastodon, it seems entirely possible to implement and could work like this:

1. Alice write a post and locks replies from people she doesn't follow
2. Beatrice, who is not followed by Alice, then cannot reply. Her client app or server won't allow to write and submit a reply.

But what if Beatrice uses a modified "bad" client or server?
1. Beatrice's reply is stored on her bad server
2. Beatrice's server notify Alice's server that a reply was sent
3. Alice's server knows that Alice's post should be locked, and simply disregard the notification.
4. Alice's server never shows the reply from Beatrice. Alice never knows about its existence.
5. People who follow Alice won't see Beatrice's answer, unless they are also on Beatrice's server. This way, the undesired answer is not propagated to other nice servers.

Adding more details about point # 5:

Imagine a third person, Clara, who is on another server (not the same as Alice or Beatrice). Clara's server is un-modified and "plays nice with the rules.
1. Clara is following both Alice and Beatrice
2. Clara's server would therefore receive Beatrice's undesired message, the one that was sent without Alice's consent because Beatrice uses a "bad" server.
3. Clara's server knows that Alice's post should be locked, and simply disregard the notification from Beatrice's "bad" server.
4. Clara's server deletes Beatrice's reply from its database, because it was malicious. This way, the undesired answer is not propagated to other nice servers.

**Reply by bobwyman** [reply]
While it would be technically possible to have servers reject replies to some posts, it seems to me that since the post will have a URL, someone could still link to that URL in a new post of their own. The effect would be to comment on the post, but it would potentially happen without the original poster knowing about it. This is the same situation that causes some folk to object to "Quote Posts."

I think you may be able to block replies, but you can't stop people from commenting on something you've posted.

**Reply by unusualevent** [reply]
I mean, that kinda feels like saying "blocking people from your profile won't work". Sure, you can put in effort to evade it? The difference to a blocklist and this (an allowlist) is that it's much harder to evade when the server on which the person who restricts it resides, has decided to not give the abuser power.
if push comes to shove, setting a profile to private is another option, or a post to private, sure. This is a soft gradient between those two options whose security control is a little stronger than playing block whackamole, and weaker than setting an entire profile to private and letting an abuser silence you.

**Reply by narFnarF** [reply]
@bobwyman True. But if that person is not playing nicely, others will notice and will eventually block that person (or the server)

@unusualevent I agree. The intention is that abuser should not silence people (aka force them to hide their profile). We should empower victims by giving them tools to protect themselves.

**Comment by narFnarF** [comment]
I updated my suggestion based on some replies: 
https://github.com/mastodon/mastodon/discussions/21444#discussioncomment-5210393

**Comment by jonpincus** [comment]
Erin Shephard has some relevant thoughts in "Right to Reply" https://blog.erinshepherd.net/2023/04/right-of-reply/

**Reply by bobwyman** [reply]
I almost agree with Erin Shephard. My one objection is the suggestion that the relevant issue is:
> "who’s replies get to show up under your post."

As Erin points out, services can and will differ in which replies "show up" under a post. Some clients, in at least some views or modes, won't present any replies at all. Some might filter or block some replies. And, some clients, when presenting replies, might include replies that aren't in the source object's Replies collection. (i.e. They may implement a rule saying that any object which references another is considered a "reply" to it. For example, if one were to "reinvent SideWiki..." or otherwise use a search engine to find references to objects.) 

Given that ActivityPub defines "Replies" as a property of Objects, I suggest that we should focus on the fact that replies, if included in "replies" properties, modify objects. This means that the question really important question should be: _Who is allowed to modify an object created by someone else?_ I believe that the creator of an object should have a privileged right to control modifications to that object. Thus, the debate shouldn't be about which replies "show up," the real issue is:
>"who's replies get to modify your post."

Someone who creates an object "inReplyTo" another object should be considered to have no more right to modify that object's "replies" property than they would have to modify any other property of that object. One has no right to modify the "content" or "to" properties of someone else's objects. Why would anyone have a right to modify the "replies" property?

Certainly, object creators should be able to adopt rules such as "Allow all InReplyTo's to modify Replies." It would even be reasonable to state that this policy is a very useful default. However, object creators should also have the right to restrict modifications to their objects -- with the understanding that servers and clients will retain the right to display whatever they may wish when listing replies.

**Comment by narFnarF** [comment]
Nearly a year later, is there an update on this?

**Comment by lash0000** [comment]
Any updates in regardless to this? everyone out there at Fediverse or ActivityHub?

**Comment by rtng** [comment]
I use the reply function mainly for my own posts. In this way, I post on certain topics and use the thread structure for this. I think in advance about the order in which I post certain media and texts on the respective topic. And this is exactly where I think the option to prevent comments from other users for these topic-related threads is very useful, because even though I am actually happy about the reactions of other users, their comments cause confusion at exactly this point, so that other users can no longer follow the actual course of the topic-related thread properly.

Elsewhere, I am very happy about the comments of other users. In addition, I also make topics where I literally ask for the participation and contributions of other users, unfortunately the participation is quite low for these topics, but I do not give up hope.

In summary, my wish would be to have the possibility and to be able to decide whether I leave the comment function for the respective post/thread or suppress it, but of course the author him/her/self must retain the possibility to continue /her/his thread.

FYI:
Here an example for a topic related thread "Gravitational Lensing" https://defcon.social/@grobi/114374350096488478
